pkg/datastore/datastore_with_txcontext_test.go

```diff
@@ -126,17 +129,91 @@ func BenchmarkTxCrud(b *testing.B) {
 	LOG = logger.WithField(datastore.COMP, datastore.SAAS_PERSISTENCE)
 
 	var t testing.T
-	ds, _ := SetupDataStore("BenchmarkCrud")
+	ds, _ := SetupDataStore("BenchmarkTxCrud")
 	defer ds.Reset()
-	myCokeApp, user1, user2 := SetupDbTables(ds)
+	_, user1, user2 := SetupDbTables(ds)
 	for n := 0; n < b.N; n++ {
-		testCrud(&t, ds, CokeAdminCtx, myCokeApp, user1, user2)
+		testTxCrud(&t, ds, CokeAdminCtx, user1, user2)
 	}
 }
 
 func TestTxCrud(t *testing.T) {
 	ds, _ := SetupDataStore("TestTxCrud")
 	defer ds.Reset()
-	myCokeApp, user1, user2 := SetupDbTables(ds)
-	testTxCrud(t, ds, CokeAdminCtx, myCokeApp, user1, user2)
+	_, user1, user2 := SetupDbTables(ds)
+	testTxCrud(t, ds, CokeAdminCtx, user1, user2)
+}
+
+func TestSingleTxCrud(t *testing.T) {
+	assert := assert.New(t)
+	var err error
+
+	ctx := CokeAdminCtx
+	ds, _ := SetupDataStore("TestSingleTxCrud")
+	defer ds.Reset()
+	_, user1, user2 := SetupDbTables(ds)
+
+	txFetcher := authorizer.SimpleTransactionFetcher{}
+	tx, err := ds.Helper().GetDBTransaction(ctx, datastore.GetTableName(user1), user1)
+	assert.NoError(err)
+	defer rollbackTx(t, tx)
+	txCtx := txFetcher.WithTransactionCtx(ctx, tx)
+
+	t.Log("Querying created records in same transaction")
+	for _, record := range []*AppUser{user1, user2} {
+		queryResult := AppUser{Id: record.Id}
+		err = ds.Find(txCtx, &queryResult)
+		assert.NoError(err)
+		assert.Equal(record, &queryResult)
+	}
+
+	t.Log("Verifying update and find in same transaction")
+	user1.Name = "Jeyhun G."
+	user1.Email = "jeyhun111@mail.com"
+	user1.EmailConfirmed = !user1.EmailConfirmed
+	user1.NumFollowers++
+	user2.Name = "Jahangir G."
+	user2.Email = "jahangir111@mail.com"
+	user2.EmailConfirmed = !user2.EmailConfirmed
+	user2.NumFollowers--
+	tx, err = ds.Helper().GetDBTransaction(ctx, datastore.GetTableName(user1), user1)
```

Comment:
Same

Please use ds.GetTransaction() to init transaction. ds.Helper().GetDBTransaction() (or ds.GetDBTransaction()) is used in every CRUD function to either extract the transaction pointer out of ctx, or init a brand new one

Reply:
Done